libs/extensions/std/exec/src/archive-interpreter-executor.ts

```diff
@@ -83,7 +82,12 @@ export class ArchiveInterpreterExecutor
             mimeType,
             content,
           );
-          root.putFile(relPath, file);
+          if (root.putFile(relPath, file) == null) {
+            return R.err({
+              message: `File with name ${file.name} and path ${relPath} failed to insert`,
+              diagnostic: { node: context.getCurrentNode(), property: 'name' },
+            });
+          }
```

Comment:
Better use `assert` here, because we expect that all `putFile` calls succeed at runtime:
```suggestion
          const addedFile = root.putFile(relPath, file);
          assert(addedFile != null);
```